### PR TITLE
add an expectedFailure test component

### PR DESCRIPTION
**Pull request by Jessica-Koch**
add the expected failure test component for use in cyprus tests

### CODE REVIEW

File: apps/teams-test-app/src/components/AppInitialization.tsx [hunk 1]
```diff
@@ -45,12 +45,35 @@ const NotifyFailure = (): React.ReactElement =>
...
+    onClick: {
+      validateInput: input => {
+        if (!input) {
+          // this API actually allow for the input not to be provided
+          return;
+        }
+        const acceptableValues = Object.values(app.ExpectedFailureReason);
+        if (!acceptableValues.includes(input)) {
+          throw new Error(`input must be one of: ${JSON.stringify(acceptableValues)}`);
+        }
```
Comment (anmaur) [inline apps/teams-test-app/src/components/AppInitialization.tsx:61]: input should be of type below to align with the API "IExpectedFailureRequest " :

File: apps/teams-test-app/src/components/AppInitialization.tsx [hunk 2]
```diff
@@ -45,12 +45,35 @@ const NotifyFailure = (): React.ReactElement =>
     },
   });
 
+const NotifyExpectedFailure = (): React.ReactElement =>
+  ApiWithTextInput<app.ExpectedFailureReason>({
+    name: 'appInitializationExpectedFailure2',
```
Comment (anmaur) [inline apps/teams-test-app/src/components/AppInitialization.tsx:50]: nit : '2', We can get rid of 2 since its the first time we are writing tests for this

**Comment by erinha** [comment]
<p>Analyzed commit id : de09c6bf6fae98e5837ba65827a5d4e7baeef9c8</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 84d8ca3a551852fd526398b97237c56eca7cd2d6</p>

File: apps/teams-test-app/src/components/AppInitialization.tsx (hunk 2 above)
Comment (Jessica-Koch) [reply apps/teams-test-app/src/components/AppInitialization.tsx:50]: will do!

File: apps/teams-test-app/src/components/AppInitialization.tsx (hunk 1 above)
Comment (Jessica-Koch) [reply apps/teams-test-app/src/components/AppInitialization.tsx:61]: I apologize, I'm not sure what you mean by "type below", this is probably very obvious.  I am pulling the values from the IExpectedFailure enum

**Comment by erinha** [comment]
<p>Analyzed commit id : c4580431442f6c023180a10396c31332ceb0cf60</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 84d8ca3a551852fd526398b97237c56eca7cd2d6</p>

File: apps/teams-test-app/src/components/AppInitialization.tsx (hunk 1 above)
Comment (anmaur) [reply apps/teams-test-app/src/components/AppInitialization.tsx:61]: My bad for vague comment above  I did not realize type did not turn up there.
The type of notifyExpectedFailure is :   
export interface IExpectedFailureRequest {
    reason: ExpectedFailureReason;
    message?: string;
  }

Comment (Jessica-Koch) [reply apps/teams-test-app/src/components/AppInitialization.tsx:61]: ok this makes sense after talking offline...

**Comment by erinha** [comment]
<p>Analyzed commit id : 39cad2e66e0f7b9758517167cc16b415cc097669</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 84d8ca3a551852fd526398b97237c56eca7cd2d6</p>

**Review by MarcinJuraszek — APPROVED** [review]

**Pull request merged**